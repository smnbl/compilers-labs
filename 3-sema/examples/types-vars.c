// RUN-WITH-ARGS: -dump-type-table

int variables_typecheck(int i, float f, string s)
{
    i;
    f;
    s;

    float foo[2];

    foo[0];
}
