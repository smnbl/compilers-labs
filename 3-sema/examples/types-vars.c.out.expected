└── Program <16>
    └── FuncDecl: returnType = 'int', name = 'variables_typecheck' <15>
        ├── VarDecl: type = 'int', name = 'i' <0>
        ├── VarDecl: type = 'float', name = 'f' <1>
        ├── VarDecl: type = 'string', name = 's' <2>
        └── CompoundStmt <14>
            ├── ExprStmt <4>
            │   └── VarRefExpr: name = 'i' <3>
            ├── ExprStmt <6>
            │   └── VarRefExpr: name = 'f' <5>
            ├── ExprStmt <8>
            │   └── VarRefExpr: name = 's' <7>
            ├── ArrayDecl: type = 'float', name = 'foo' <10>
            │   └── IntLiteral: value = '2' <9>
            └── ExprStmt <13>
                └── ArrayRefExpr: name = 'foo' <12>
                    └── IntLiteral: value = '0' <11>
Type table:
0                   i64
1                   float
2                   i8*
3                   i64
5                   float
9                   i64
7                   i8*
10                  [2 x float]
11                  i64
12                  float
